Kernel bug fixes

vbnogueira opened the pull request.

vbnogueira committed:
p4tc: Set constant entry pointer to NULL after freeing it

Set constant entry pointer to NULL after freeing it, so that the next
dump command that sees if doesn't accidentally use the freed pointer.

Fixes: 2df57c838bc9

Signed-off-by: Victor Nogueira <victor@mojatatu.com>

vbnogueira committed:
p4tc: Merge uapi file p4tc_entities.h into p4tc.h

To make debian happy and help in the vagrant build, lets merge
p4tc_entities.h into p4tc.h. This is also simplify the uapi header
files by removing one of them.

Fixes: fab05bc5732d

Signed-off-by: Jamal Hadi Salim <jhs@mojatatu.com>

vbnogueira committed:
p4tc: Avoid capital letters in EXTRAVERSION

Debian doesn't enjoy capital letters in kernel versions. So, to make
debian happier, let's avoid this.

Fixes: 3f78e25d35f9

Signed-off-by: Jamal Hadi Salim <jhs@mojatatu.com>

vbnogueira committed:
p4tc: Fix #if CONFIG_NET_P4_TC to #ifdef CONFIG_NET_P4_TC in skbuff.c

The #if CONFIG_NET_P4_TC that guards the definition of function
skb_ext_put_p4tc should be an #ifdef instead.

Fixes: 235490dc90d7

Signed-off-by: Jamal Hadi Salim <jhs@mojatatu.com>

vbnogueira committed:
p4tc: Remove any mention of metact from MAINTAINERS file

Remove any remaining mention of metact from MAINTAINERS file.

Fixes: 97fe48e2df18

Signed-off-by: Victor Nogueira <victor@mojatatu.com>

vbnogueira committed:
p4tc: Check if tb[P4TC_ENTRY_TBLNAME] is NULL before calling tcf_table_find_byany

When tb[P4TC_ENTRY_TBLNAME] is NULL, nla_data(tb[P4TC_ENTRY_TBLNAME])
may return something other then NULL. This means we should check
tb[P4TC_ENTRY_TBLNAME]'s content before passing it to tcf_table_find_byany.

Fixes: fab05bc5732d

Signed-off-by: Victor Nogueira <victor@mojatatu.com>

vbnogueira committed:
ci/cd: Remove mention of metact from s390x config file

Remove legacy metact config (CONFIG_NET_ACT_METACT) from the s390x config
file.

Fixes: 03e43cda8105

Signed-off-by: Victor Nogueira <victor@mojatatu.com>

jhsmt reviewed (changes requested):
You are missing the changes from yesterday to add support for p4tcgdb

vbnogueira committed:
ci/cd: Remove mention of metact from x86 config file and adapt config to p4tcgdb

Remove legacy metact config (CONFIG_NET_ACT_METACT) from the x86 config
file. Also adapt the config file to include all the requirements needed for
p4tcgdb to work in the near future.

Fixes: 7e7d29d

Signed-off-by: Victor Nogueira <victor@mojatatu.com>

vbnogueira commented:
> p4tcgdb

Done here: https://github.com/p4tc-dev/linux-p4tc-pub/pull/43/commits/928e4263bf9a20134ab17f60e1ea9f672841fa7e

tammela reviewed (approved):
LGTM
